core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -345,16 +345,22 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) extends NonEmptyCollec
   override def distinctBy[B](f: A => B)(implicit O: Order[B]): NonEmptyList[A] = {
     implicit val ord: Ordering[B] = O.toOrdering
```

Comment:
actually this could be moved inside the `else` branch to avoid it on singletons.